packages/@react-spectrum/s2/src/SelectBoxGroup.tsx

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright 2025 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {
+  ContextValue,
+  ListBox,
+  ListBoxItem
+} from 'react-aria-components';
+import {DOMRef, DOMRefValue, Orientation, Selection} from '@react-types/shared';
+import {getAllowedOverrides, StyleProps} from './style-utils' with {type: 'macro'};
+import React, {createContext, forwardRef, ReactElement, ReactNode, useEffect, useMemo} from 'react';
+import {SelectBoxRenderPropsContext} from './SelectBox';
+import {style} from '../style' with {type: 'macro'};
+import {useControlledState} from '@react-stately/utils';
+import {useDOMRef} from '@react-spectrum/utils';
+import {useSpectrumContextProps} from './useSpectrumContextProps';
+
+export interface SelectBoxGroupProps extends StyleProps {
```

Comment:
you might want to extend from `ListBoxProps` from RAC and omit any prop you don't want, or override any props that you might want to change the type of (like selectionMode for example)